browser/brave_wallet/brave_wallet_service_delegate_impl_android.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_BRAVE_WALLET_BRAVE_WALLET_SERVICE_DELEGATE_IMPL_ANDROID_H_
+#define BRAVE_BROWSER_BRAVE_WALLET_BRAVE_WALLET_SERVICE_DELEGATE_IMPL_ANDROID_H_
+
+#include <memory>
+#include <string>
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
+
+#include "brave/components/brave_wallet/browser/brave_wallet_service_delegate.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+
+namespace content {
+class BrowserContext;
+}
+
+namespace brave_wallet {
+
+class ExternalWalletsImporter;
+
+class BraveWalletServiceDelegateImpl : public BraveWalletServiceDelegate {
```

Comment:
why do we have BraveWalletServiceDelegateImpl here instead of using `BraveWalletProviderDelegateImpl`? https://github.com/brave/brave-core/pull/9443/files#diff-534f2c78996c290a0f72fda008f21cf5795dbfc6d9221381e0b2b7e54491dfe7R40

Reply:
`BraveWalletProviderDelegateImpl` and `BraveWalletServiceDelegateImpl` are for different purposes and delegates for different classes